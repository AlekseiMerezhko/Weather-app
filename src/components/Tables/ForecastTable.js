import React, { useState } from "react";
import Switch from "@material-ui/core/Switch";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Moment from "react-moment";

const ForecastTable = ({ currentCity, dailyForecast }) => {
  const [state, setState] = useState({
    isFahrenheit: true
  });

  const handleChange = name => event => {
    setState({ ...state, [name]: event.target.checked });
  };

  const fToC = fahrenheit => {
    const fToCel = ((fahrenheit - 32) * 5) / 9;
    return Math.round(fToCel);
  };

  return (
    <div>
      <FormControlLabel
        control={
          <Switch
            checked={state.isFahrenheit}
            onChange={handleChange("isFahrenheit")}
            value="isFahrenheit"
          />
        }
        label="Fahrenheit"
      />

      <table className="table-auto">
        <thead>
          <tr>
            <th className="px-4 py-2">City</th>
            <th className="px-4 py-2">Date</th>
            <th className="px-4 py-2">
              Minimum Temperature {state.isFahrenheit ? "째F" : "째C"}:
            </th>
            <th className="px-4 py-2">
              Maximum Temperature {state.isFahrenheit ? "째F" : "째C"}:
            </th>
            <th className="px-4 py-2">Day</th>
            <th className="px-4 py-2">Night</th>
          </tr>
        </thead>
        {dailyForecast && currentCity ? (
          <tbody>
            {dailyForecast.map(day => (
              <tr key={day.Link}>
                <td className="border px-4 py-2">
                  {currentCity.LocalizedName}
                </td>
                <td className="border px-4 py-2">
                  <Moment format="MM/DD/YYYY">{day.Date}</Moment>
                </td>
                <td className="border px-4 py-2">
                  {state.isFahrenheit
                    ? day.Temperature.Minimum.Value
                    : fToC(day.Temperature.Minimum.Value)}
                </td>
                <td className="border px-4 py-2">
                  {state.isFahrenheit
                    ? day.Temperature.Maximum.Value
                    : fToC(day.Temperature.Maximum.Value)}
                </td>
                <td className="border px-4 py-2">{day.Day.IconPhrase}</td>
                <td className="border px-4 py-2">{day.Night.IconPhrase}</td>
              </tr>
            ))}
          </tbody>
        ) : null}
      </table>
    </div>
  );
};

export default ForecastTable;
