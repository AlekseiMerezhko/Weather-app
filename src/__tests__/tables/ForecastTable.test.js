import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import ForecastTable from "../../components/Tables/ForecastTable";

describe("<ForecastTable />", () => {
  it("Check ForecastTable render", () => {
    const footer = renderer.create(<ForecastTable />).toJSON();
    expect(footer).toMatchSnapshot();
  });
});
