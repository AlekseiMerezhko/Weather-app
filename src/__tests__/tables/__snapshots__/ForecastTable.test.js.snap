// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Check ForecastTable render 1`] = `
<div>
  <label
    className="MuiFormControlLabel-root"
  >
    <span
      className="MuiSwitch-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-19 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-20 Mui-checked"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={true}
            className="PrivateSwitchBase-input-22 MuiSwitch-input"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
            value="isFahrenheit"
          />
          <span
            className="MuiSwitch-thumb"
          />
        </span>
      </span>
      <span
        className="MuiSwitch-track"
      />
    </span>
    <span
      className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
    >
      Fahrenheit
    </span>
  </label>
  <table
    className="table-auto"
  >
    <thead>
      <tr>
        <th
          className="px-4 py-2"
        >
          City
        </th>
        <th
          className="px-4 py-2"
        >
          Date
        </th>
        <th
          className="px-4 py-2"
        >
          Minimum Temperature 
          °F
          :
        </th>
        <th
          className="px-4 py-2"
        >
          Maximum Temperature 
          °F
          :
        </th>
        <th
          className="px-4 py-2"
        >
          Day
        </th>
        <th
          className="px-4 py-2"
        >
          Night
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`<Footer /> properly increments and decrements the counter 1`] = `
<div>
  <label
    className="MuiFormControlLabel-root"
  >
    <span
      className="MuiSwitch-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-19 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-20 Mui-checked"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={true}
            className="PrivateSwitchBase-input-22 MuiSwitch-input"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
            value="isFahrenheit"
          />
          <span
            className="MuiSwitch-thumb"
          />
        </span>
      </span>
      <span
        className="MuiSwitch-track"
      />
    </span>
    <span
      className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
    >
      Fahrenheit
    </span>
  </label>
  <table
    className="table-auto"
  >
    <thead>
      <tr>
        <th
          className="px-4 py-2"
        >
          City
        </th>
        <th
          className="px-4 py-2"
        >
          Date
        </th>
        <th
          className="px-4 py-2"
        >
          Minimum Temperature 
          °F
          :
        </th>
        <th
          className="px-4 py-2"
        >
          Maximum Temperature 
          °F
          :
        </th>
        <th
          className="px-4 py-2"
        >
          Day
        </th>
        <th
          className="px-4 py-2"
        >
          Night
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`<ForecastTable /> Check ForecastTable render 1`] = `
<div>
  <label
    className="MuiFormControlLabel-root"
  >
    <span
      className="MuiSwitch-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-19 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-20 Mui-checked"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={true}
            className="PrivateSwitchBase-input-22 MuiSwitch-input"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
            value="isFahrenheit"
          />
          <span
            className="MuiSwitch-thumb"
          />
        </span>
      </span>
      <span
        className="MuiSwitch-track"
      />
    </span>
    <span
      className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
    >
      Fahrenheit
    </span>
  </label>
  <table
    className="table-auto"
  >
    <thead>
      <tr>
        <th
          className="px-4 py-2"
        >
          City
        </th>
        <th
          className="px-4 py-2"
        >
          Date
        </th>
        <th
          className="px-4 py-2"
        >
          Minimum Temperature 
          °F
          :
        </th>
        <th
          className="px-4 py-2"
        >
          Maximum Temperature 
          °F
          :
        </th>
        <th
          className="px-4 py-2"
        >
          Day
        </th>
        <th
          className="px-4 py-2"
        >
          Night
        </th>
      </tr>
    </thead>
  </table>
</div>
`;
